emcc.py

```diff
@@ -955,7 +948,7 @@ def check(input_file):
       newargs = CC_ADDITIONAL_ARGS + newargs
 
       if separate_asm and final_suffix != 'html':
-        logging.warning("--separate-asm works best when compiling to HTML. otherwise, you must yourself load the '.asm.js' file that is emitted separately, and must do so before loading the main '.js` file")
+        shared.WarningManager.warn(shared.WarningType.SEPARATE_ASM, "--separate-asm works best when compiling to HTML. Otherwise, you must yourself load the '.asm.js' file that is emitted separately, and must do so before loading the main '.js' file. Pass in '-Wno-warn-separate-asm' to disable this warning.")
```

Comment:
To fit in even more with clang, we can use clang's format for displaying the warning flag, e.g.
` warning: --separate-asm works best ... '.js' file [-Wwarn-separate-asm]`. You could even put that inside of `WarningManager.warn` so you wouldn't have to repeat it here.